framework/src/media/PlayerWorker.cpp

```diff
@@ -46,7 +46,8 @@ int PlayerWorker::entry()
 			if (mCurPlayer && (mCurPlayer->mCurState == PLAYER_STATE_PLAYING)) {
 				shared_ptr<Decoder> mDecoder = mCurPlayer->mInputDataSource->getDecoder();
 				if(mDecoder) {
-					size_t num_read = mCurPlayer->mInputDataSource->read(mCurPlayer->mBuffer, std::min(mCurPlayer->mBufSize, mDecoder->getDataSpace()));
+					size_t num_read = mCurPlayer->mInputDataSource->read(mCurPlayer->mBuffer,
```

Comment:
I think this logic should be moved to inputDataSource, will you do that later??
```
if (mCurPlayer && (mCurPlayer->mCurState == PLAYER_STATE_PLAYING)) {
    size_t num = mCurPlayer->mInputDataSource->read(mBuffer, mBufsize);
    if (num > 0) {
         playback();
    }
}

size_t read(unsigned char* buf, size_t size) {
    if ((mAudioType != AUDIO_TYPE_PCM) && (mDecoder != NULL)) {
            pushData();
            getFrame();
    } else {
            fread();
    }
}
```

Reply:
You're right. I have to modify the logic in another PR.